1679 Add icon support to va-accordion-item

mdewey opened the pull request:
## Chromatic
<!-- This `1679-accordion-header-update` is a placeholder for a CI job - it will be updated automatically -->
https://1679-accordion-header-update--60f9b557105290003b387cd5.chromatic.com


## Description
Closes [Ticket](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1679)
Address UX updates needed for [secure messaging](https://www.sketch.com/s/32bdb345-fb1f-409b-985f-ff627cb60032)
## Testing done
- [ ] Chrome
- [x] Storybook
- [ ] Various Screen sizes
- [x] updated unit testing

## Screenshots
<img width="281" alt="image" src="https://user-images.githubusercontent.com/1793923/233200863-faca4e72-39a2-4dde-a489-76d09d384ac5.png">
<img width="289" alt="image" src="https://user-images.githubusercontent.com/1793923/233200913-01eaf841-50b5-4153-afde-780a52d59f81.png">



## Acceptance criteria
- [ ] Icons and bolding are in the header

## Definition of done
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

## Important notes

- I am looking for feedback on the approach, [context](https://github.com/department-of-veterans-affairs/component-library/pull/674#discussion_r1164625345), before fixing the failing tests, polishing the code and polishing the docs. 
- I am still working on the docs page and cleaning up the samples. but wanted to get the review process started while I do that.

jamigibbs commented on packages/web-components/src/components/va-accordion/va-accordion-item.tsx:
```diff
@@ -92,41 +93,41 @@ export class VaAccordionItem {
+            aria-controls="content"
+            style={{ fontWeight: boldness }}
+          >
+            {< slot name="icon" />}
```
Just because it's a named slot doesn't mean it's limited to just icons unless we programmatically only allow icon nodes to be passed into it but if @humancompanion-usds is fine with this approach, I don't have any issues with it. I think it makes sense to have it a slot since we're unsure of which icon library would be used here in the future.

Minor style question: Is it necessary to have `{ }` around the slot?

humancompanion-usds replied:
I defer to you both. Seems fine to me.

mdewey replied:
Minor style answer: Those can be removed, It was a ternary thing there before, but that turned out not to be needed.

jamigibbs commented on packages/web-components/src/components/va-accordion/va-accordion.tsx:
```diff
@@ -70,13 +70,8 @@ export class VaAccordion {
-      event.detail.currentTarget.parentNode.parentNode.host || // if we are on IE, `.host` won't be there
-      event.detail.currentTarget.parentNode.parentNode;
+    // eslint-disable-next-line i18next/no-literal-string
+    const clickedItem = (event.target as HTMLElement).closest('va-accordion-item');
```
Is this essentially just an IE cleanup/refactor?

mdewey replied:
Basically. On that topic, what browsers do we need to support?

jamigibbs replied:
> that browsers do we need to support?

I forgot to answer your actual question! Here is our current [browser support policy](https://vfs.atlassian.net/wiki/spaces/pilot/pages/1847951366/Browser+Support+Policy):

![Screenshot 2023-04-20 at 10 10 53 AM](https://user-images.githubusercontent.com/872479/233409503-e4d8fdb5-a958-47e8-b9e8-963c5d16c508.png)

jamigibbs commented on packages/web-components/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@department-of-veterans-affairs/web-components",
-  "version": "4.33.1",
+  "version": "4.34.1",
```
If we're doing a minor bump, let's kick this to 4.34.0

jamigibbs commented on packages/web-components/src/components/va-accordion/va-accordion.tsx:
```diff
@@ -70,22 +70,22 @@ export class VaAccordion {
+    if (header) {
+      headerText = header.innerHTML
+      headerLevel = parseInt(header.tagName.toLowerCase().split('')[1]);
+    } else {
```
Any particular reason that the [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) `?` was removed in various places?

jamigibbs commented on packages/web-components/src/components/va-accordion/va-accordion.tsx:
```diff
@@ -70,22 +70,22 @@ export class VaAccordion {
-
+    // eslint-disable-next-line i18next/no-literal-string
+    const clickedItem = (event.target as HTMLElement).closest('va-accordion-item');
+    console.log(clickedItem)
```
Remove console log.

jamigibbs commented on README.md:
````diff
@@ -79,10 +86,46 @@ This will allow you to run Storybook locally to view all components
+
+```bash
+npx stencil test --e2e -- src/components/[component-name]/[component-name].e2e.ts
+```
````
Thank you (again)!

jamigibbs reviewed (commented):
A lot of nice additions here! Thanks again for your help. 

I left just a few comments and also can you double check the responsive view for long headline text? I suspect we will not want the text to wrap underneath the icon like this but I'll let @humancompanion-usds confirm:

![Screenshot 2023-04-21 at 4 24 58 PM](https://user-images.githubusercontent.com/872479/233737232-5a0294b2-158a-492e-a84f-87eb1ec84011.png)

It might be as simple as adding `display: flex` to the button:

![Screenshot 2023-04-21 at 4 31 12 PM](https://user-images.githubusercontent.com/872479/233737878-33d54cc4-260d-4021-b634-3eaee429f6a0.png)

![Screenshot 2023-04-21 at 4 33 12 PM](https://user-images.githubusercontent.com/872479/233738015-f85118ef-3b76-4fbd-99a4-6957db726efd.png)

mdewey commented:
@jamigibbs I have addressed all your comments (great catches! ). Let me know what else I need to do here.  Once this is merged I will update vets-website

jamigibbs reviewed (approved):
This looks great, thanks Mark! We will need to create a new component library release in order to make this available in vets-website which isn't a problem but before merging, could you:

1. Update the `core` package version to 15.2.0
2. Update the PR title to something more descriptive to the actual update (this will help us with the release notes). ie. Add icon support to va-accordion-item
